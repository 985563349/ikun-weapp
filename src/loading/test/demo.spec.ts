import path from 'node:path';
import simulate from 'miniprogram-simulate';

test('loading snapshot', async () => {
  const id = simulate.load(path.resolve(__dirname, '../demo/loading'), 'basic');
  const container = simulate.render(id);

  container.attach(document.createElement('parent-wrapper'));
  expect(container.toJSON()).toMatchSnapshot();
});
