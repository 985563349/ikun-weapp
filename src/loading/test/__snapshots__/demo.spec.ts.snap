// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading snapshot 1`] = `
<basic>
  <wx-view>
    <wx-view>
      Spinner 加载
    </wx-view>
    <wx-view>
      <wx-view>
        尺寸
      </wx-view>
      <wx-view>
        <k-loading
          size="sm"
        />
        <k-loading />
        <k-loading
          size="lg"
        />
      </wx-view>
    </wx-view>
    <wx-view>
      <wx-view>
        主题
      </wx-view>
      <wx-view>
        <k-loading />
        <k-loading
          color="primary"
        />
        <k-loading
          color="success"
        />
        <k-loading
          color="warning"
        />
        <k-loading
          color="danger"
        />
      </wx-view>
    </wx-view>
    <wx-view>
      <wx-view>
        标签
      </wx-view>
      <wx-view>
        <k-loading
          label="加载中..."
        />
      </wx-view>
    </wx-view>
    <wx-view>
      <wx-view>
        垂直布局
      </wx-view>
      <wx-view>
        <k-loading
          label="加载中"
          vertical="{{true}}"
        />
      </wx-view>
    </wx-view>
    <wx-view>
      <wx-view>
        标签颜色
      </wx-view>
      <wx-view>
        <k-loading
          label="加载中"
          vertical="{{true}}"
        />
        <k-loading
          color="primary"
          label="加载中"
          vertical="{{true}}"
        />
      </wx-view>
    </wx-view>
  </wx-view>
</basic>
`;
